### DISCUSSION TITLE
CardPayment Brick or initMercadoPago problem

### DISCUSSION THREAD

**Discussion by joxpulp** [mercadopago/sdk-react#34] [Q&A] [answered]
initMercadoPago does not reload if I change the publicKey, for example I have several accounts from different MercadoPago countries, the buyer selects an initial currency and the CardPayment component is shown with the document types for that country, but if he then selects another currency from another country, the CardPayment component does not reload and continues showing me the same document types of the previous country, in the initMercadoPago method I am changing the publicKey when changing currency but it does not seem to work and it continues taking the publicKey of the previous currency.

First selection: publicKey from an AR MercadoPago account:

<img width="647" alt="Captura de Pantalla 2023-04-20 a la(s) 11 12 44 a  m" src="https://user-images.githubusercontent.com/73070103/233393246-a34948a5-9124-4b32-ad45-6512e76af6c8.png">

Changed currency and publicKey for CL MercadoPago account (not changing document types):

<img width="647" alt="Captura de Pantalla 2023-04-20 a la(s) 11 12 44 a  m" src="https://user-images.githubusercontent.com/73070103/233393246-a34948a5-9124-4b32-ad45-6512e76af6c8.png">

Expected output if i change the publicKey and currency to CL MercadoPago account;

<img width="619" alt="Captura de Pantalla 2023-04-20 a la(s) 11 14 30 a  m" src="https://user-images.githubusercontent.com/73070103/233393713-92b399e6-d5a9-4bc8-b43c-669459da030e.png">

**Comment by joxpulp** [comment]
UPDATE: I found a way to achieve what I wanted: 

1. Import  `MercadoPagoInstance` class in the component or file you use `initMercadoPago` method.
2. Reset the       `MercadoPagoInstance.instanceMercadoPago` variable to `undefined`.
3. Later call again `initMercadoPago` to store a new instance in `MercadoPagoInstance.instanceMercadoPago` with the updated `publicKey`.

> initMercadoPago does not reload if I change the publicKey, for example I have several accounts from different MercadoPago countries, the buyer selects an initial currency and the CardPayment component is shown with the document types for that country, but if he then selects another currency from another country, the CardPayment component does not reload and continues showing me the same document types of the previous country, in the initMercadoPago method I am changing the publicKey when changing currency but it does not seem to work and it continues taking the publicKey of the previous currency.
> 
> First selection: publicKey from an AR MercadoPago account:
> 
> <img alt="Captura de Pantalla 2023-04-20 a la(s) 11 12 44 a  m" width="647" src="https://user-images.githubusercontent.com/73070103/233393246-a34948a5-9124-4b32-ad45-6512e76af6c8.png">
> 
> Changed currency and publicKey for CL MercadoPago account (not changing document types):
> 
> <img alt="Captura de Pantalla 2023-04-20 a la(s) 11 12 44 a  m" width="647" src="https://user-images.githubusercontent.com/73070103/233393246-a34948a5-9124-4b32-ad45-6512e76af6c8.png">
> 
> Expected output if i change the publicKey and currency to CL MercadoPago account;
> 
> <img alt="Captura de Pantalla 2023-04-20 a la(s) 11 14 30 a  m" width="619" src="https://user-images.githubusercontent.com/73070103/233393713-92b399e6-d5a9-4bc8-b43c-669459da030e.png">

**Comment by francineemilia** [comment, marked as answer]
Hi @joxpulp,

Thanks for reporting the situation and the work around you found!
With your report we opened a bug and we're working at it.
When the solution is available, we will report here again

Regards,

**Reply by joxpulp** [reply]
Thank you 😄

**Comment by ghost** [comment]
@joxpulp hello! this issue was fixed in release 0.0.9: https://github.com/mercadopago/sdk-react/releases/tag/v0.0.9
